### PR TITLE
chore: remove the deprecated PyTorch API [DET-3262]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

BREAKING CHANGE: remove the deprecated Pytorch API.

## Test Plan

N/A

## Commentary (optional)

N/A

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: examples/hp_search_benchmarks/darts_penntreebank_pytorch/model_def.py [hunk 1]
```diff
@@ -118,49 +149,6 @@ def get_genotype_from_hps(self):
             cell_config.append((edge_op, edge_ind))
         return Genotype(recurrent=cell_config, concat=range(1, 9))
 
-    def build_model(self) -> nn.Module:
```
Comment (rb-determined-ai) [inline examples/hp_search_benchmarks/darts_penntreebank_pytorch/model_def.py:121]: Can we frontload these changes into the next release?  I didn't even realize that we still had old-style expamples laying around.

File: docs/reference/api/pytorch.txt [hunk 2]
```diff
@@ -141,9 +140,9 @@ class with ``PyTorchTrial``, implement the following callback:
 .. code::
 
    class ReduceLROnPlateauEveryValidationStep(PyTorchCallback):
-       def __init__(self, context):
+       def __init__(self, optimizer):
```
Comment (rb-determined-ai) [inline docs/reference/api/pytorch.txt:143]: I have a slight preference for leaving this as `context` in order to hint users towards a more generally applicable practice.

**Review by rb-determined-ai — COMMENTED** [review]
Looks good, I'll approve it after the next release is cut (you may have to remind me)

File: docs/reference/api/pytorch.txt (hunk 2 above)
Comment (shiyuann) [reply docs/reference/api/pytorch.txt:143]: `context.get_optimizer()` will be removed. We recommend users to save their optimizers in member fields of their objects but they can't use it in the `Callback`.

**Review by rb-determined-ai — APPROVED** [review]
